### PR TITLE
PP-541 Remove old status field.

### PR DESCRIPTION
**Pull request by ijmad**
Now we have the 'state' field everywhere, we can remove
the old legacy status from the publicapi.

Improve swagger here and there.

### CODE REVIEW

**Comment by ijmad** [comment]
rebuild

**Comment by ijmad** [comment]
rebuild

File: src/main/java/uk/gov/pay/api/model/PaymentState.java [hunk 1]
```diff
@@ -4,20 +4,25 @@
...
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.JsonNode;
+import io.swagger.annotations.ApiModel;
 import io.swagger.annotations.ApiModelProperty;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @JsonIgnoreProperties(ignoreUnknown = true)
+@ApiModel(value="Payment state", description = "A structure representing the current state of the payment in its lifecycle.")
 public class PaymentState {
     @JsonProperty("status")
```
Comment (kakumara) [inline src/main/java/uk/gov/pay/api/model/PaymentState.java]: Shall we give a swagger example for status too?

File: src/main/java/uk/gov/pay/api/model/PaymentState.java [hunk 2]
```diff
@@ -4,20 +4,25 @@
...
     private String status;
 
+    @ApiModelProperty(value = "Whether the payment has finished", required = true)
```
Comment (kakumara) [inline src/main/java/uk/gov/pay/api/model/PaymentState.java]: I think the `@ApiModelProperty` annotations here are overridden by the ones in the associated annotation ion the getter method. better to remove these and use only the one around getter.

**Comment by kakumara** [comment]
Otherwise 👍

File: src/main/java/uk/gov/pay/api/model/PaymentState.java (hunk 2 above)
Comment (ijmad) [inline src/main/java/uk/gov/pay/api/model/PaymentState.java]: Sorted!

**Pull request merged**